### DISCUSSION TITLE
Copilot only returns this Error now Request Error: Stream closed with error code NGHTTP2_INTERNAL_ERROR

### DISCUSSION THREAD

**Discussion by dtlevin** [community/community#55575] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

Copilot was working great all last week. It stopped working today. All we get is this error over and over again. Even if we have it write a simple CSS class.  We are signed in to Github with no firewalls and no VPN, and nothing has changed since we started to use it. 

Does anyone know this issue, or is it on Github's end? Is there any support for paid products?

**Comment by nickfmc** [comment]
been seeing this off an on today also

**Comment by Akash1134** [comment]
Hi @dtlevin and @nickfmc ,

We're sorry to hear that Copilot is not working as expected for you.

I'd like to know a couple of things:

- Do you have more than one account linked to Copilot via your IDE?
- Are you using the latest version of Copilot and VS Code (or any other IDE)?

If you are also using Copilot Nightly along with Copilot in your IDE, please remove it and try using the stable release of the Copilot extension.

Thanks!

**Comment by nickfmc** [comment]
not using the nightly no, and only one github account used. The only thing that I do that might not be standard use is that I was coding via the VCS remote-SSH extension to another device on my network. I have since added copilot to the remote computers VSC, I'll see if that makes any difference.

**Comment by sergiyvelichkevych** [comment]
Have the same issue with VCS remote-SSH. It worked several days ago for sure and now I have
[INFO] [default] [2023-05-17T17:27:07.921Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[ERROR] [default] [2023-05-17T17:27:17.981Z] Request Error: Stream closed with error code NGHTTP2_INTERNAL_ERROR
[ERROR] [ghostText] [2023-05-17T17:27:17.983Z] Error on ghost text request FetchError: Stream closed with error code NGHTTP2_INTERNAL_ERROR

**Reply by berkeleymalagon** [reply]
same error here. complete unusable. this is a paid subscription, this must be fixed asap

**Comment by ronuraltindag** [comment]
I have the same problem, co-pilot stopped working with no reason on all my devices. All of them give the same error. I appreciate any help. 

![image](https://github.com/community/community/assets/10189326/651ccdf7-7374-4a83-8ec7-af43a7aa5b9b)

**Reply by nickfmc** [reply]
@ronuraltindag are you using the VSC remote-ssh extension also?

**Reply by ronuraltindag** [reply]
Yes. It started to function again after the day I posted the issue.

**Comment by JonathanSL** [comment]
Same issue using a Dev Container

**Comment by davecheney** [comment]
Thank you for your report. We're investigating. The short version is, the model took too long to think of an answer so the request timed out, you can check that by noting the time between request and failure -- 10 seconds.

The `NGHTTP2_INTERNAL_ERROR` error code is misleading, but there isn't a lot we can do about that right now as it comes from the guts of the node.js runtime.

**Reply by jackbackes** [reply]
Any result to investigation? I started receiving this error on both nightly and stable today.

**Reply by jackbackes** [reply]
to be clear, my use of Copilot is COMPLETELY BROKEN now. Perhaps a new timeout was introduced? It often takes more than 10 seconds to get a response from the server, so if a 10 second timeout was introduced that will break a lot of queries.

**Reply by jackbackes** [reply]
confirmed that it still works on small files where the server responds more quickly than 10 seconds. My request is that timeout is increased to something more reasonable, like 45 seconds.

**Reply by jackbackes** [reply]
> there isn't a lot we can do about that right now as it comes from the guts of the node.js runtime

I have to believe timeouts are configurable, no?

**Reply by davecheney** [reply]
@jackbackes i'm sorry you are experiencing such difficulty. I strongly recommend you contact support who can escalate this issue to the development team; https://support.github.com/contact.

**Reply by sergiyvelichkevych** [reply]
Why did you mark the question as answered ? Please reopen the ticket, because I do not see any other way to post bug.

**Reply by berkeleymalagon** [reply]
This has broken my entire development workflow and rendered copilot useless. Please advise how to configure the timeout

**Comment by sergiyvelichkevych** [comment]
The problem is still there. Copilot is not working. And we still pay subscription. What is the plan to fix the issue?

**Reply by sovetski** [reply]
I also have the same problem, my file is not even 700 lines, it's not correct for a paid tool

**Reply by alexis-coulombe** [reply]
Same here, even with an empty file i still have the error.

**Reply by esbozos** [reply]
same here.

**Reply by davecheney** [reply]
@sergiyvelichkevych looking at your telemetry I see only one request from your account in the past 7 days. Does that match your experience?

**Comment by berkeleymalagon** [comment]
I'm having this same issue currently. Copilot has become completely unusable. This is a paid sub so I'm expecting a fix ASAP

**Reply by davecheney** [reply]
@berkeleymalagon i'm very sorry you are having difficulties. I've checked your telemetry and in the last 3 days I see no incidents of this specific problem. Can you please confirm the github username you are accessing copilot with?

**Reply by berkeleymalagon** [reply]
It was resolved itself. im using the extension remotely via ssh host in vscode. it failed for a few hours today and then resolved itself

**Reply by davecheney** [reply]
Thank you. It may be possible that what you experienced was unrelated, `NGHTTP2_INTERNAL_ERROR` is a catch all error code that node.js uses. It can occur in multiple circumstances. 

If the problem re-occurs, I recommend contacting support who can assist you in collecting debug information and escalate to the development team if required.

**Comment by davecheney** [comment, marked as answer]
Thank you for your patience. We have deployed a fix and believe the issue is resolved.

**Reply by jigsnptel** [reply]
Hi, i am not able to resolve this error. and this error occirs when i am working with php file. pls help

**Reply by EgorkZe** [reply]
The same for me, not working at all

**Reply by avp1598** [reply]
Same not working for me today

**Reply by Prymethron** [reply]
Seems like problem started again, also same problem occurs.

**Reply by kamblepratik90** [reply]
yes noticing this again now.

**Comment by samerkassem82** [comment]
I am having this error right now, I have tried different internet providers too.

**Comment by ozzzzz** [comment]
Got another error, probably that is related to this topic:
```
{"error":{"code":"429","message":"Requests to the Create a completion from a chosen model Operation under OpenAI Language Model Instance API have exceeded call rate limit of your current OpenAI S0 pricing tier. Please retry after 1 second. Please go here: https://aka.ms/oai/quotaincrease if you would like to further increase the default rate limit.","policy_id":"ResourceRateLimitBy1s"}}
```
